projects/igniteui-angular/src/lib/grids/filtering/grid-filtering.service.ts

```diff
@@ -53,14 +56,25 @@ export class IgxFilteringService implements OnDestroy {
     private column;
     private lastActiveNode;
 
-    constructor(private _moduleRef: NgModuleRef<any>, private iconService: IgxIconService,  private _overlayService: IgxOverlayService) {}
+    private get componentInstance() {
+        return this.grid.excelStyleFilteringComponent ??
+            this.createComponentInstance(IgxGridExcelStyleFilteringComponent);
```

Comment:
Creating the component instance in this getter is error-prone. We should call `createComponentInstance` only before attach.